src/Microsoft.AspNetCore.BuildTools.ApiCheck/ApiListingComparer.cs

```diff
@@ -13,128 +13,108 @@ public class ApiListingComparer
     {
         private readonly ApiListing _newApiListing;
         private readonly ApiListing _oldApiListing;
-        private readonly IList<ApiChangeExclusion> _exclusions;
 
         public ApiListingComparer(
             ApiListing oldApiListing,
-            ApiListing newApiListing,
-            IList<ApiChangeExclusion> exclusions = null)
+            ApiListing newApiListing)
         {
             _oldApiListing = oldApiListing;
             _newApiListing = newApiListing;
-            _exclusions = exclusions ?? new List<ApiChangeExclusion>();
         }
 
-        public ApiComparisonResult GetDifferences()
+        public IList<BreakingChange> GetDifferences()
         {
-            var exclusions = _exclusions.ToList();
-
             var breakingChanges = new List<BreakingChange>();
+            var newTypes = _newApiListing.Types;
+
             foreach (var type in _oldApiListing.Types)
             {
                 var newType = _newApiListing.FindType(type.Name);
-                if (newType == null || !string.Equals(type.Id, newType.Id, StringComparison.Ordinal))
+                if (newType == null)
                 {
-                    var isAcceptable = newType != null && IsAcceptableTypeChange(type, newType);
-                    if (!isAcceptable)
+                    breakingChanges.Add(new BreakingChange(type.Id, /*memberId:*/ null, ChangeKind.Removal));
+                }
+                else
+                {
+                    newTypes.Remove(newType);
+
+                    if (!string.Equals(type.Id, newType.Id, StringComparison.Ordinal)
+                        && !IsAcceptableTypeChange(type, newType))
                     {
-                        var typeChange = FilterExclusions(type, member: null, exclusions: exclusions);
-                        if (typeChange != null)
-                        {
-                            breakingChanges.Add(typeChange);
-                        }
+                        breakingChanges.Add(new BreakingChange(type.Id, /*memberId:*/ null, ChangeKind.Removal));
```

Comment:
Sure. in EF we use commented out if its in middle and non-commented if its last. I will change this.